ui/src/theme.js

```diff
@@ -291,12 +305,21 @@ const createTheme = isDarkTheme => {
         },
       },
       MuiTableCell: {
+        head: {
+          fontSize: '0.75rem',
+        },
         root: {
           borderBottom: `1px solid ${
             isDarkTheme ? THEME.TEN_PERCENT_WHITE : THEME.TEN_PERCENT_BLACK
           }`,
           whiteSpace: 'nowrap',
         },
+        sizeSmall: {
+          padding: '4px 32px 4px 24px',
+        },
+        body: {
+          fontSize: '0.8125rem',
+        },
```

Comment:
OK, I think I was able to fix the table stuff by:

* Removing tableText class name
* Adding a prop `size` to ConnectionDataTable (we need a similar logic to DataTable as well)
* Make TableCellItem stop using a List and instead just use flexbox (The long term is to remove this component and just have a class defined in theme.js, but that can be done later)
* Adding size="medium" to tables that don't have any kind of buttons, links
* Removing `<Typography variant="body2">...</Typgraphy>` and replacing it with `...` (only the ones with variant body2) since Material-ui adds the body2 variant style for things that are in a table.

```diff
--- a/ui/src/components/CachePurgesTable/index.jsx
+++ b/ui/src/components/CachePurgesTable/index.jsx
@@ -86,6 +86,7 @@ export default class CachePurgesTable extends Component {
 
     return (
       <ConnectionDataTable
+        size="medium"
         connection={sortedCachePurgesConnection}
         pageSize={VIEW_CACHE_PURGES_PAGE_SIZE}
         sortByHeader={sortBy}
diff --git a/ui/src/components/ClientsTable/index.jsx b/ui/src/components/ClientsTable/index.jsx
index 2bb565f10..646ea5bca 100644
--- a/ui/src/components/ClientsTable/index.jsx
+++ b/ui/src/components/ClientsTable/index.jsx
@@ -4,11 +4,8 @@ import { pipe, map, sort as rSort } from 'ramda';
 import memoize from 'fast-memoize';
 import { camelCase } from 'change-case/change-case';
 import TableRow from '@material-ui/core/TableRow';
-import { withStyles } from '@material-ui/core/styles';
 import TableCell from '@material-ui/core/TableCell';
 import LinkIcon from 'mdi-react/LinkIcon';
-import { ListItemText } from '@material-ui/core';
-import Typography from '@material-ui/core/Typography';
 import TableCellItem from '../TableCellItem';
 import ConnectionDataTable from '../ConnectionDataTable';
 import DateDistance from '../DateDistance';
@@ -23,11 +20,6 @@ const sorted = pipe(
 );
 const tableHeaders = ['Client ID', 'Last Date Used'];
 
-@withStyles({
-  tableText: {
-    fontSize: '0.8125rem',
-  },
-})
 export default class ClientsTable extends Component {
   static propTypes = {
     clientsConnection: shape({
@@ -83,7 +75,7 @@ export default class ClientsTable extends Component {
   };
 
   render() {
-    const { onPageChange, clientsConnection, classes } = this.props;
+    const { onPageChange, clientsConnection } = this.props;
     const { sortBy, sortDirection } = this.state;
     const iconSize = 16;
 
@@ -105,16 +97,12 @@ export default class ClientsTable extends Component {
             <TableCell>
               <Link to={`/auth/clients/${encodeURIComponent(client.clientId)}`}>
                 <TableCellItem button>
-                  <ListItemText>
-                    <Typography className={classes.tableText}>
-                      {client.clientId}
-                    </Typography>
-                  </ListItemText>
+                  {client.clientId}
                   <LinkIcon size={iconSize} />
                 </TableCellItem>
               </Link>
             </TableCell>
-            <TableCell className={classes.tableText}>
+            <TableCell>
               <DateDistance from={client.lastDateUsed} />
             </TableCell>
           </TableRow>
diff --git a/ui/src/components/ConnectionDataTable/index.jsx b/ui/src/components/ConnectionDataTable/index.jsx
index b9d39e16a..1740219fe 100644
--- a/ui/src/components/ConnectionDataTable/index.jsx
+++ b/ui/src/components/ConnectionDataTable/index.jsx
@@ -54,6 +54,7 @@ export default class ConnectionDataTable extends Component {
     headers: null,
     onHeaderClick: null,
     withoutTopPagination: false,
+    size: 'small',
   };
 
   static propTypes = {
@@ -109,6 +110,10 @@ export default class ConnectionDataTable extends Component {
      * at the top of the table.
      */
     withoutTopPagination: bool,
+    /**
+     * Allows TableCells to inherit size of the Table.
+     */
+    size: oneOf(['small', 'medium']),
   };
 
   state = {
@@ -214,6 +219,7 @@ export default class ConnectionDataTable extends Component {
       sortByHeader,
       sortDirection,
       withoutTopPagination,
+      size,
     } = this.props;
     const { count } = this.getPaginationMetadata();
     const colSpan = columnsSize || (headers && headers.length) || 1;
@@ -222,7 +228,7 @@ export default class ConnectionDataTable extends Component {
       <Fragment>
         {!withoutTopPagination && this.renderTablePagination(colSpan, count)}
         <div className={classes.tableWrapper}>
-          <Table>
+          <Table size={size}>
             {headers && (
               <TableHead>
                 <TableRow
diff --git a/ui/src/components/TableCellItem/index.jsx b/ui/src/components/TableCellItem/index.jsx
index 6d5442b27..8334769bc 100644
--- a/ui/src/components/TableCellItem/index.jsx
+++ b/ui/src/components/TableCellItem/index.jsx
@@ -18,6 +18,8 @@ import ListItem from '@material-ui/core/ListItem';
   },
   listItemButton: {
     ...theme.mixins.listItemButton,
+    display: 'flex',
+    justifyContent: 'space-between',
   },
 }))
 /**
diff --git a/ui/src/theme.js b/ui/src/theme.js
index 38084c29d..503a9e4ad 100644
--- a/ui/src/theme.js
+++ b/ui/src/theme.js
@@ -61,11 +61,6 @@ const createTheme = isDarkTheme => {
   };
 
   return {
-    props: {
-      MuiTableCell: {
-        size: 'small',
-      },
-    },
     palette: {
       type: isDarkTheme ? 'dark' : 'light',
       background: {
@@ -306,21 +301,12 @@ const createTheme = isDarkTheme => {
         },
       },
       MuiTableCell: {
-        head: {
-          fontSize: '0.75rem',
-        },
         root: {
           borderBottom: `1px solid ${
             isDarkTheme ? THEME.TEN_PERCENT_WHITE : THEME.TEN_PERCENT_BLACK
           }`,
           whiteSpace: 'nowrap',
         },
-        sizeSmall: {
-          padding: '4px 32px 4px 24px',
-        },
-        body: {
-          fontSize: '0.8125rem',
-        },
       },
       MuiPickersToolbar: {
         toolbar: {

```

Reply:
Wow! It looks great! Thank you, I will finish it up!))